ocaml/xapi/xapi_vm_lifecycle.ml

```diff
@@ -875,6 +883,8 @@ let force_state_reset_keep_current_operations ~__context ~self ~value:state =
     Db.VM.set_resident_on ~__context ~self ~value:Ref.null ;
     (* make sure we aren't reserving any memory for this VM *)
     Db.VM.set_scheduled_to_be_resident_on ~__context ~self ~value:Ref.null ;
+    if not (Db.VM.get_is_a_template ~__context ~self) then
```

Comment:
Could this be cleared together with `RestartVM`?

Reply:
Yes, it can be cleared with no problem, but it will generate useless logs...

Reply:
I understand here is to clear `RestartDeviceModel` at the point of VM stop and suspend. Would the domid changing point capture them already?

Reply:
`domid` is changed in below line:
`Db.VM.set_domid ~__context ~self ~value:(-1L) ;`